packages/dss-bridge/script/lib/XDomainDss.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.14;
+
+import { Vm } from "forge-std/Vm.sol";
+import { Cure } from "xdomain-dss/Cure.sol";
+import { Dai } from "xdomain-dss/Dai.sol";
+import { DaiJoin } from "xdomain-dss/DaiJoin.sol";
+import { End } from "xdomain-dss/End.sol";
+import { Pot } from "xdomain-dss/Pot.sol";
+import { Jug } from "xdomain-dss/Jug.sol";
+import { Spotter } from "xdomain-dss/Spotter.sol";
+import { Vat } from "xdomain-dss/Vat.sol";
+
+struct DssInstance {
+    Vat vat;
+    Dai dai;
+    DaiJoin daiJoin;
+    Spotter spotter;
+    Pot pot;
+    Jug jug;
+    Cure cure;
+    End end;
+}
+
+interface AuthLike {
+    function rely(address) external;
+    function deny(address) external;
+}
+
+// Tools for deploying and setting up an xdomain-dss instance
+library XDomainDss {
+
+    function switchOwner(address base, address newOwner) internal {
+        AuthLike(base).rely(newOwner);
+        AuthLike(base).deny(address(this));
+    }
+
+    function deploy(address pauseProxy) internal returns (DssInstance memory dss) {
```

Comment:
Is it really the pauseProxy or should be the gov relay guest?

Reply:
Yes was going to change this.